### ISSUE TITLE
Virtual library still not works under dune 3.15

### ISSUE THREAD

**Issue by hackwaly** [melange-re/melange#1105]
dune version: 3.15
ocaml version: 5.1.1
melange version: both 3.0.0-51 and [df3c2ce](https://github.com/melange-re/melange/commit/df3c2ce403de10cbfe58365c541327abaaae7195) reproduced

A miminal reproducible demo is uploaded here: 

https://github.com/hackwaly/minimal-melange-virtual-library

```
File "common/StringBuilder.ml", line 1:
Error: Pal__StringBuilder not found, it means either the module does not exist or it is a namespace
```

**Comment by anmonteiro** [comment]
Thanks for trying this out- I’ll take a look!

**Comment by anmonteiro** [comment]
Thanks a lot for your test case. I managed to reproduce your issue here: https://github.com/ocaml/dune/pull/10410

**Comment by hackwaly** [comment]
Shall this to be released in dune 3.15.1?

**Comment by anmonteiro** [comment]
I'll ask in the Dune PR.
